utils/config.go

```diff
@@ -278,16 +278,13 @@ func EnsureConfigFile(fileName string) (string, error) {
 	return "", fmt.Errorf("no config file found")
 }
 
-// LoadConfig will try to search around for the corresponding config file.  It will search
-// /tmp/fileName then attempt ./config/fileName, then ../config/fileName and last it will look at
-// fileName.
-func LoadConfig(fileName string) (config *model.Config, configPath string, appErr *model.AppError) {
+func loadConfigFile(fileName string) (config *model.Config, configPath string, appErr *model.AppError) {
 	if fileName != filepath.Base(fileName) {
 		configPath = fileName
 	} else {
 		if path, err := EnsureConfigFile(fileName); err != nil {
 			appErr = model.NewAppError("LoadConfig", "utils.config.load_config.opening.panic", map[string]interface{}{"Filename": fileName, "Error": err.Error()}, "", 0)
-			return
+			return nil, "", appErr
```

Comment:
If we're not using the named parameters in the function signature anymore, should we remove them?

Reply:
Ah, I missed the purpose of named parameters.  Been some time I've done Go dev, I may have missed some best practices 😓 

I like the documentation's explanation of having named parameters as a way of documenting your code
https://golang.org/doc/effective_go.html#named-results
> The names are not mandatory but they can make code shorter and clearer: they're documentation. If we name the results of nextInt it becomes obvious which returned int is which.

Either way, this does avoid us from initializing the variables so it's not completely pointless.
If we want to take full advantage of the named parameter feature, do you prefer returning without explicitly stating the results?

Basically `return` instead of `return nil, "", appErr` <- that can get accomplished by the named result parameters feature

Reply:
I would prefer keeping the named parameters. that would provide a cleaner solution

I'll revet back to only `returning`

Reply:
I know there's lots of strong opinions on named parameters, so I'm 0/5 on this at the moment, but to share from https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters:

>  Don't name result parameters just to avoid declaring a var inside the function; that trades off a minor implementation brevity at the cost of unnecessary API verbosity.

and 

> Naked returns are okay if the function is a handful of lines. Once it's a medium sized function, be explicit with your return values. Corollary: it's not worth it to name result parameters just because it enables you to use naked returns. Clarity of docs is always more important than saving a line or two in your function.